### PR TITLE
User story/800 as a user, i want to start otvision from the command line using a config yaml

### PR DESCRIPTION
**Pull request by randy-seng**
OP#800

### CODE REVIEW

File: OTVision/config.py [hunk 1]
```diff
@@ -19,12 +19,527 @@
...
+
+
+@dataclass(frozen=True)
+class _Filetypes:
+    video_filetypes: _VideoFiletypes = _VideoFiletypes()
+    image_filetypes: _ImageFiletypes = _ImageFiletypes()
+    detect: str = ".otdet"
+    track: str = ".ottrk"
+    refpts: str = ".otrfpts"
+    transform: str = ".gpkg"
```
Comment (briemla) [inline OTVision/config.py:99]: Can we use _DefaultFileType.detect, ...track, ...refpts here?

File: OTVision/config.py [hunk 2]
```diff
@@ -19,12 +19,527 @@
...
+            "CALIBRATIONS": self.calibrations,
+            "REFPTS": self.refpts,
+        }
+
+
+@dataclass
+class _ConvertConfig:
+    paths: list[Path] = field(default_factory=list)
+    run_chained: bool = True
+    output_filetype: str = ".mp4"
```
Comment (briemla) [inline OTVision/config.py:134]: Can we use _DefaultFileType.video or _VideoFileType.mp4 here?

File: OTVision/config.py [hunk 3]
```diff
@@ -19,12 +19,527 @@
...
+    yolov5l: str = "yolov5l"
+    yolov5x: str = "yolov5x"
+
+    def to_list(self) -> list:
+        return [self.yolov5s, self.yolov5m, self.yolov5l, self.yolov5x]
+
+
+@dataclass
+class _YoloConfig:
+    weights: str = "yolov5s"
```
Comment (briemla) [inline OTVision/config.py:183]: Can we use `_YoloWeights.yolov5s` here?

File: OTVision/config.py [hunk 4]
```diff
@@ -19,12 +19,527 @@
...
+    video: str = ".mp4"
+    image: str = ".jpg"
+    detect: str = ".otdet"
+    track: str = ".ottrk"
+    refpts: str = ".otrfpts"
+
+    @staticmethod
+    def from_dict(d: dict) -> "_DefaultFiletype":
+        return _DefaultFiletype(
+            d.get("VID", _DefaultFiletype.video),
```
Comment (briemla) [inline OTVision/config.py:41]: Might it be useful to replace `"VID"` with a constant and use the constant at every place `"VID"` is used? 

Same for all the other parameter ids.

**Review by briemla — DISMISSED** [review]
Looks nice only a few comments.

**Review by briemla — APPROVED** [review]

**Review by martinbaerwolff — APPROVED** [review]

**Pull request merged**